app/views/components/_select-with-search.html.erb

```diff
@@ -0,0 +1,36 @@
+<%
+  id ||= false
+  label ||= false
+  name ||= id
+  is_page_heading ||= false
+
+  shared_helper = GovukPublishingComponents::Presenters::SharedHelper.new(local_assigns)
+  heading_size = false unless shared_helper.valid_heading_size?(heading_size)
+  select_helper = GovukPublishingComponents::Presenters::SelectWithSearchHelper.new(local_assigns)
+
+  aria_describedby = { describedby: select_helper.describedby }
+%>
+
+<%= content_tag :div, class: select_helper.css_classes, data: { module: "select-with-search" } do %>
```

Comment:
I think we need an option to pass in data modules on init. For example we track selection and deselection of supporting and lead orgs on the edit page with the `track-select-click` module

Reply:
I've added support for tracking in 6feb67404f083c7148accc268620db626b33099c (2nd commit on this PR)

The `track-select-click` module can't be used directly because the `<select>` element is not preserved once Choices.js has kicked in. So I've gone with an implementation similar to the one used on the Autocomplete component (see #7013).